server/utils/kvstore/kvstore.go

```diff
@@ -56,3 +59,46 @@ func StoreJSON(s KVStore, key string, v interface{}) (returnErr error) {
 	}
 	return s.Store(key, data)
 }
+
+func AtomicModify(s KVStore, key string, modify func(initialValue []byte) ([]byte, error)) error {
+	readModify := func() ([]byte, []byte, error) {
+		initialBytes, appErr := s.Load(key)
+		if appErr != nil {
+			return nil, nil, errors.Wrap(appErr, "unable to read inital value")
+		}
+
+		modifiedBytes, err := modify(initialBytes)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "modification error")
+		}
+
+		return initialBytes, modifiedBytes, nil
+	}
+
+	var (
+		retryLimit     = 5
+		retryWait      = 30 * time.Millisecond
```

Comment:
May be better here, but wondering why not putting this on the top of the file as constants.